apps/src/lab2/views/dialogs/GenericDialog.tsx

```diff
@@ -64,54 +65,68 @@ import moduleStyles from './generic-dialog.module.scss';
  * If no confirm button text is provided, the default text is "OK" (translatable).
  */
 
-type UseClosingCallbackArgs = {
+export type GetButtonCallbackArgs = {
   closeDialog: DialogCloseFunctionType;
   closeType: DialogCloseActionType;
   callback: dialogCallback | undefined;
   disabled: boolean | undefined;
 };
 
-const useClosingCallback = ({
-  closeDialog,
-  closeType,
-  callback,
-  disabled,
-}: UseClosingCallbackArgs) =>
-  useCallback(() => {
+export const defaultGetButtonCallback =
+  ({closeDialog, closeType, callback, disabled}: GetButtonCallbackArgs) =>
+  () => {
     if (!disabled) {
       closeDialog(closeType);
       callback && callback();
     }
-  }, [closeDialog, closeType, callback, disabled]);
+  };
+
+const useButtonCallback = ({
+  closeDialog,
+  closeType,
+  callback,
+  disabled,
+  getButtonCallback,
+}: GetButtonCallbackArgs & {
+  getButtonCallback: typeof defaultGetButtonCallback;
+}) =>
+  useMemo(
+    () => getButtonCallback({closeDialog, closeType, callback, disabled}),
+    [closeDialog, closeType, callback, disabled, getButtonCallback]
+  );
```

Comment:
In order to continue to memoize the callback, we needed an extra layer of indirection.

So we add a second function which will return the function that will generate the button callback. _That_ function can be handed in as a prop. And then inside of the `GenericDialog`, we memoize the result of _that_ function.

If we attempted to hand in a memoized function, we'd violate the rules of hooks with a conditional.

Honestly, considering how low impact the dialog is, it may be worth just circling back and removing the memoization entirely and just letting the thing re-render as the callbacks change. Most of the time, it'd only draw during initial paint anyway. But we can circle back to that.